va_explorer/va_analytics/dash_apps/va_dashboard.py

```diff
@@ -851,32 +1007,77 @@ def get_metric_display_names(map_metrics):
 
 
 # ====================Geographic Levels (View options)============#
-@app.callback(
-    [
-        Output(component_id="view_level", component_property="options"),
-        Output(component_id="view_level", component_property="disabled"),
-    ],
-    [
-        Input(component_id="filter_dict", component_property="data"),
-        Input(component_id="location_types", component_property="data"),
-    ],
+# callback 6
+# @app.callback(
+#     [
+#         Output(component_id="view_level", component_property="options"),
+#         Output(component_id="view_level", component_property="disabled"),
+#     ],
+#     [
+#         Input(component_id="filter_dict", component_property="data"),
+#         Input(component_id="location_types", component_property="data"),
+#     ],
+# )
+# def update_view_options(filter_dict, location_types, **kwargs):
+#     START_TIME = time.time()
+#     if filter_dict is not None:
+
+#         # only activate when user is zoomed out and hasn't searched for anything
+#         disable = (filter_dict["geo_filter"] or filter_dict["search_filter"])
+#         if not disable:
+#             view_options = location_types
+#            # label_class = "input-label"
+#         else:
+#             view_options = []
+#             #label_class = "input-label-disabled"
+#         options = [{"label": o.capitalize(), "value": o} for o in view_options]
+#         print("update view Callback", time.time() - START_TIME)
+#         return options, disable #, label_class
+
+### Conversion of update_view_options to client-side callbacks ###
+### have to use two different callbacks since we can only have one output in each (django-plotly-dash constraint)
+app.clientside_callback(
+    """
+    function(filter_dict){
+        if(filter_dict){
+            // console.log(filter_dict);
+            return((filter_dict['geo_filter'] === true) || (filter_dict['search_filter'] === true));
+        }
+        return null;
+    }
+    """,
+    Output(component_id="view_level", component_property="disabled"),
+    [Input(component_id="filter_dict", component_property="data")]
+
+)
+
+app.clientside_callback(
+    """
+    function(disable, location_types){
+        
+            if(disable === false){
+                view_options = location_types;
+            }
+            else{
+                view_options = [];
+            }
+            options = [];
+            for(var i =0; i<view_options.length;i++){
+                options.push({"label":view_options[i].charAt(0).toUpperCase() + view_options[i].slice(1), "value":view_options[i]})
```

Comment:
this is a personal preference, but I would declare `view_options[i].charAt(0).toUpperCase() + view_options[i].slice(1)` to a helpful variable name, since its a bit long of a function

Reply:
Good call. Updated.